### PR TITLE
meta: move Snap's from_dict() system-username parsing into SystemUser

### PR DESCRIPTION
**Pull request by cjp256**
Tidy it up a bit in preparation for further cleanup of from_dict().

No functional changes.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: tests/unit/meta/test_system_user.py [hunk 1]
```diff
@@ -71,3 +71,73 @@ def test_invalid_user_non_shared_scope_from_dict(self):
...
+            errors.SystemUsernamesValidationError,
+            SystemUser.from_object,
+            user_name=user_name,
+            user_object=user_object,
+        )
+
+        self.assertThat(
+            error.get_brief(),
+            Equals(
+                "Invalid system-usernames for 'name': scope 'invalid_scope' is invalid"
```
Comment (sergiusens) [inline tests/unit/meta/test_system_user.py]: let's please keep message string comparisons on test_errors

Comment (cjp256) [reply tests/unit/meta/test_system_user.py]: I can do that, though test_errors really just ensures the error class is correct.

The assertRaises checks the correct type.

The string check ensures correct usage of the init parameters, though that could be done on the object fields too (i.e. check error._name, error._message). Or just skip altogether?

Comment (cjp256) [reply tests/unit/meta/test_system_user.py]: nvm, see "check for attributes" below. I concur

Comment (cjp256) [reply tests/unit/meta/test_system_user.py]: done

**Pull request merged**